### PR TITLE
Tryout tower

### PR DESCRIPTION
**Pull request by DonTars**
Please check the scene thoroughly, I merged it from the def branch, it worked for me.

### CODE REVIEW

**Review by spirit126 — COMMENTED** [review]
Tested and works fine. No Unit Tests added.

File: Assets/Scripts/AutoShooter.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+using UnityEngine;
+
+public class AutoShooter : MonoBehaviour
+{
+    public bool shootsPlayer;
+    public Shooter shooter;
+    private GameObject toBeShot;
+    
+    // Update is called once per frame
+    void Update()
```
Comment (spirit126) [inline Assets/Scripts/AutoShooter.cs:13]: make method private

File: Assets/Scripts/AutoShooter.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        toBeShot = shootsPlayer ? SearchPlayer() : FindClosestEnemy();
+
+        if (toBeShot != null)
+        {
```
Comment (spirit126) [inline Assets/Scripts/AutoShooter.cs:18]: use LINQ (is not null)

File: Assets/Scripts/SpawnTower.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+public class SpawnTower : MonoBehaviour
+{
+    
+    public GameObject objToSpawn;
+    public Transform objSpawnPlace;
+    
+    private float myTime = 0F;
+    private float nextPlacement = 1F;
+    private float placementRate = 1F;
```
Comment (spirit126) [inline Assets/Scripts/SpawnTower.cs:14]: Make these variables accessible from the outside? ([SerializeField])

File: Assets/Scripts/SpawnTower.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
```
Comment (spirit126) [inline Assets/Scripts/SpawnTower.cs:20]: remove empty method

**Review by spirit126 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/AutoShooter.cs (hunk 1 above)
Comment (DonTars) [reply Assets/Scripts/AutoShooter.cs]: changed accordingly

File: Assets/Scripts/SpawnTower.cs (hunk 3 above)
Comment (DonTars) [reply Assets/Scripts/SpawnTower.cs]: changed accordingly

File: Assets/Scripts/SpawnTower.cs (hunk 4 above)
Comment (DonTars) [reply Assets/Scripts/SpawnTower.cs]: changed accordingly

**Pull request closed without merging**